src/main/java/com/exasol/releasedroid/progress/Estimation.java

```diff
@@ -0,0 +1,118 @@
+package com.exasol.releasedroid.progress;
+
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Objects;
+
+// [impl->dsn~estimate-duration~1]
+public class Estimation {
+
+    /**
+     * @return estimation with undefined duration
+     */
+    public static Estimation empty() {
+        return new Estimation(null, null);
+    }
+
+    /**
+     * @param duration duration representing the estimation
+     * @return new instance of {@link Estimation}
+     */
+    public static Estimation of(final Duration duration) {
+        return new Estimation(null, duration);
+    }
+
+    /**
+     * Compute estimation defined by period of time between start and end.
+     *
+     * @param start start of time period to compute estimation from
+     * @param end   end of time period to compute estimation from
+     * @return duration representing the estimation
+     */
+    public static Estimation from(final Date start, final Date end) {
+        final Instant startInstant = start.toInstant();
+        return new Estimation(startInstant, Duration.between(startInstant, end.toInstant()));
+    }
+
+    private Instant timestamp;
+    private Duration duration;
+
+    Estimation(final Instant timestamp, final Duration duration) {
+        this.timestamp = timestamp;
+        this.duration = duration;
+    }
+
+    /**
+     * @return time stamp reference of estimation
+     */
+    public Instant timestamp() {
+        return this.timestamp;
+    }
+
+    /**
+     * @return duration of estimation
+     */
+    public Duration duration() {
+        return this.duration;
+    }
+
+    /**
+     * @return {@code true} if current estimation has non-null duration and time stamp
+     */
+    public boolean isPresent() {
+        return (this.duration != null) && (this.timestamp != null);
+    }
+
+    /**
+     * @param other another estimation to be added to the current one
+     * @return this for fluent programming
+     */
+    public Estimation add(final Estimation other) {
+        if (other.duration != null) {
+            this.duration = durationOrZero().plus(other.duration);
+            if (this.timestamp == null) {
+                this.timestamp = other.timestamp();
+            }
+        }
+        return this;
+    }
```

Comment:
Create & return a new instance to make this class immutable.

Reply:
👍